### ISSUE TITLE
Checkboxes in filter area are not visible when tabbing through the Map page

### ISSUE THREAD

**Issue by siggrid** [ushahidi/platform#4115]
**Describe the bug**
Users can’t see and so check/uncheck checkboxes in the filter area, when using a keyboard to navigate the page. Checkboxes are skipped, even though they receive focus.

**Where was the bug observed**
localhost:3000/views/map

**To Reproduce**
1. Go to Map page
2. Use Tab key to navigate through page
3. See error

**Expected behavior**
When using keyboard user can see and interact with checkboxes in filter area

**Screenshots**
Current behaviour 
![hover-only](https://user-images.githubusercontent.com/52042098/95978690-2efe6400-0e23-11eb-9cde-0f551258a78d.gif)

**URL / Environment where this happened**
local environment 

**Desktop Hardware Details (please complete the following information):**
 - Browser: Chrome

**Additional context**
Due to styling the checkbox becomes visible only on hover state of parent element.
Possible solution: in addition to :hover use  :focus-within pseudo-class. So when elements inside the parent element receive focus, required styles are applied. 
![focus-within](https://user-images.githubusercontent.com/52042098/95978827-65d47a00-0e23-11eb-9d98-321dc980bb74.gif)

Drawback of this solution: focus-within pseudo-class is not supported by IE
Also, need to go through Data Source checkboxes and labels styles, because they aren’t shown in the same way as Survey filter checkboxes and labels (while markup seems to be equal)

**Comment by Ifycode** [comment]
@siggrid Have you started work on this? Please let me know. Thanks.

**Comment by Ifycode** [comment]
> @Angamanga The thing is I only edited the code for issue #4115 #4116 in the dev tool and not in the actual code base. I found that some of the Issues also happens on the map and search area of the page. I suggest that since @siggrid found the other issue first, if she has started work on it, then let me work on this issue I just raised instead.

**Comment by siggrid** [comment]
> @siggrid Have you started work on this? Please let me know. Thanks.

Hi, @Ifycode Yes, I have. Draft result is on second gif, need to check other styles to be sure, that everything works fine.

**Comment by Ifycode** [comment]
Oh, ok. You are doing a great job already then. @Angamanga @rowasc With your approval, can I work on #4119 instead?

**Comment by Angamanga** [comment]
@Ifycode I assigned you to #4119 👍

**Comment by siggrid** [comment]
Hi, @Angamanga Seems like I need your guidance here. Could you please share your opinion on the following?

Generally, the problem is in css-styles (checkbox visible only when hovering the parent element). To solve it we need to make changes in the Pattern library.

The first solution that comes to mind is focus-within pseudo-class, which is unfortunately not supported by Internet Explorer. Polyfill is required to support IE.
Alternative solution - create and apply new css-classes to a hoverable element and handle focus inside the hoverable element with javascript. Does it have to be a single js file in the Pattern library, if so?

Besides that, when I was testing a client with an updated pattern library, I detected another problem. Within Datasource filters focus leaves the form, when checking/unchecking checkboxes, whereas within Survey filters focus works just like it should (checkbox stays focused). Looks like the problem somehow related to the form, still working on it.

**Comment by siggrid** [comment]
**Update 2**
Came up with a solution where `ng-focus` and `ng-blur` are applied to input in order to style parent element. But still unable to fix focus problem within Data source filters.

**Comment by rowasc** [comment]
hey @siggrid , thanks ! @lfycode has also sent a solution around checkboxes, thinking it may be a good idea to coordinate between you so it looks consistent across the app https://github.com/ushahidi/platform/issues/4119 
@Angamanga I'm quite lost here, can you review th messages by @siggrid above? Thanks

**Comment by siggrid** [comment]
> hey @siggrid , thanks ! @Lfycode has also sent a solution around checkboxes, thinking it may be a good idea to coordinate between you so it looks consistent across the app #4119

Agree, I’ll text @Ifycode to make sure that our changes are not conflicting. As I have understood from a description so far, @Ifycode  currently working on an outline color issue and this one is mostly related to keyboard navigation within a page.

**Comment by siggrid** [comment]
@Angamanga This issue looked like a platform-pattern-library problem at first sight. When I started working on it, it turned out that changes needed to be done in platform-client repo as well. I submitted PR with an angular-based solution and updated previous post with an alternative.

#4115
